### PR TITLE
[RW-12896][risk=no] Converted help-sidebar.spec.tsx

### PR DESCRIPTION
**Pull request by evrii**
---
**PR checklist**

- [x] I have included an issue ID or "no ticket" in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [x] I have included a risk tag of the form `[risk=no|low|moderate|severe]` in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [ ] I have manually tested this change and my testing process is described above.
- [ ] This change includes appropriate automated tests, and I have documented any behavior that cannot be tested with code.
- [ ] I have added explanatory comments where the logic is not obvious.
- One or more of the following is true:
  - [ ] This change is intended to complete a JIRA story, so I have checked that all AC are met for that story.
  - [ ] This change fixes a bug, so I have ensured the steps to reproduce are in the Jira ticket or provided above.
  - [ ] This change impacts deployment safety (e.g. removing/altering APIs which are in use), so I have documented the impacts in the description.
  - [ ] This change includes a new feature flag, so I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later.
  - [ ] This change modifies the UI, so I have taken screenshots or recordings of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P).
  - [ ] This change modifies API behavior, so I have run the relevant E2E tests locally because API changes are not covered by our PR checks.
  - [ ] None of the above apply to this change.

### CODE REVIEW

File: ui/src/app/components/help-sidebar-icons.tsx
Comment (evrii) [inline ui/src/app/components/help-sidebar-icons.tsx:1]: Added title to genomic extraction icons for accessibility and for testability.

File: ui/src/app/components/help-sidebar.spec.tsx [hunk 1]
```diff
@@ -136,34 +128,82 @@ describe('HelpSidebar', () => {
   let runtimeStub: RuntimeApiStub;
   let appsStub: AppsApiStub;
   let props: {};
+  let user;
 
   const component = async () => {
-    const c = mountWithRouter(
-      <SWRConfig value={{ provider: () => new Map() }}>
+    /*
+    The useSWR hook in useGenomicExtractionJobs is causing
```
Comment (jmthibault79) [inline ui/src/app/components/help-sidebar.spec.tsx:135]: oof.  good catch.

Comment (jmthibault79) [reply ui/src/app/components/help-sidebar.spec.tsx:135]: Should we file a tech debt ticket to remove `swr` ?  looks like it's only used here.

File: ui/src/app/components/help-sidebar.spec.tsx [hunk 2]
```diff
@@ -136,34 +128,82 @@ describe('HelpSidebar', () => {
...
 
-  const runtimeStatusIcon = (wrapper, exists = true) => {
-    const icon = wrapper
-      .find({ 'data-test-id': 'runtime-status-icon-container' })
-      .find('svg');
-    expect(icon.exists()).toEqual(exists);
-    return icon;
+  const findRuntimeStatusIcon = async (status: RuntimeStatus) => {
+    let label: RegExp;
+    switch (status) {
```
Comment (jmthibault79) [inline ui/src/app/components/help-sidebar.spec.tsx:150]: I typically prefer `switchCase()` when the action to take is variable assignment

File: ui/src/app/components/help-sidebar.spec.tsx [hunk 3]
```diff
@@ -136,34 +128,82 @@ describe('HelpSidebar', () => {
...
+      case RuntimeStatus.RUNNING:
+        label = /icon indicating environment is running/i;
+        break;
+      case RuntimeStatus.DELETING:
```
Comment (jmthibault79) [inline ui/src/app/components/help-sidebar.spec.tsx:154]: is this correct?

File: ui/src/app/components/help-sidebar.spec.tsx [hunk 4]
```diff
@@ -225,133 +300,127 @@ describe('HelpSidebar', () => {
...
     props = { pageKey: 'cohortBuilder' };
-    const wrapper = await component();
+    component();
     currentCohortCriteriaStore.next([criteria1, criteria2]);
-    await waitForFakeTimersAndUpdate(wrapper);
-    expect(
-      wrapper.find({ 'data-test-id': 'criteria-count' }).first().props()
-        .children
-    ).toBe(2);
+    expect(await screen.findByText('2')).toBeInTheDocument();
```
Comment (jmthibault79) [inline ui/src/app/components/help-sidebar.spec.tsx:387]: correct but incomplete I think - how do we know that this is the right "2" ?

File: ui/src/app/components/help-sidebar.spec.tsx [hunk 5]
```diff
@@ -360,66 +429,59 @@ describe('HelpSidebar', () => {
...
     });
-    const wrapper = await component();
-    await waitForFakeTimersAndUpdate(wrapper);
+    component();
 
-    expect(runtimeStatusIcon(wrapper).prop('style').color).toEqual(
-      colors.asyncOperationStatus.stopping
-    );
+    expect(
+      await findRuntimeStatusIcon(RuntimeStatus.STOPPING)
```
Comment (jmthibault79) [inline ui/src/app/components/help-sidebar.spec.tsx:468]: why STOPPING and not DELETING?

File: ui/src/app/components/help-sidebar.spec.tsx [hunk 6]
```diff
@@ -436,14 +498,13 @@ describe('HelpSidebar', () => {
...
 
-    expect(runtimeStatusIcon(wrapper).prop('style').color).toEqual(
-      colors.asyncOperationStatus.error
-    );
+    within(
+      await screen.findByTestId('runtime-status-icon-container')
+    ).getByTitle('Icon indicating environment has encountered an error');
   });
-
+  //
```
Comment (jmthibault79) [inline ui/src/app/components/help-sidebar.spec.tsx:507]: rm

File: ui/src/app/components/help-sidebar.spec.tsx [hunk 7]
```diff
@@ -461,12 +522,11 @@ describe('HelpSidebar', () => {
...
     ];
-    const wrapper = await component();
-    await waitForFakeTimersAndUpdate(wrapper);
+    component();
 
-    expect(extractionStatusIcon(wrapper).prop('style').color).toEqual(
-      colors.asyncOperationStatus.starting
-    );
+    expect(
+      await findRuntimeStatusIcon(RuntimeStatus.RUNNING)
```
Comment (jmthibault79) [inline ui/src/app/components/help-sidebar.spec.tsx:528]: the previous version wanted `starting` - do you know why?

**Review by jmthibault79 — COMMENTED** [review]
A few questions, but mostly looks good

File: ui/src/app/components/help-sidebar.spec.tsx (hunk 1 above)
Comment (evrii) [reply ui/src/app/components/help-sidebar.spec.tsx:135]: Thanks :-)

I made the tech debt ticket: https://precisionmedicineinitiative.atlassian.net/browse/RW-12935

File: ui/src/app/components/help-sidebar.spec.tsx (hunk 3 above)
Comment (evrii) [reply ui/src/app/components/help-sidebar.spec.tsx:154]: I believe so. Were you expecting a different icon or one that explicitly says deleting?

File: ui/src/app/components/help-sidebar.spec.tsx (hunk 4 above)
Comment (evrii) [reply ui/src/app/components/help-sidebar.spec.tsx]: Good call. I made this more specific.

File: ui/src/app/components/help-sidebar.spec.tsx (hunk 5 above)
Comment (evrii) [reply ui/src/app/components/help-sidebar.spec.tsx]: I believe that it should be DELETING. I updated it. Thanks for pointing it out.

File: ui/src/app/components/help-sidebar.spec.tsx (hunk 7 above)
Comment (evrii) [reply ui/src/app/components/help-sidebar.spec.tsx:533]: I'm not sure, but the the test title mentions looking for the running icon. Possibly a typo. It looks like it hadn't changed meaningfully in 3 years.

File: ui/src/app/components/help-sidebar.spec.tsx (hunk 3 above)
Comment (jmthibault79) [reply ui/src/app/components/help-sidebar.spec.tsx]: yeah - this is surprising.  Maybe the label should be "stopping or deleting" if it's the same icon

File: ui/src/app/components/help-sidebar.spec.tsx (hunk 7 above)
Comment (jmthibault79) [reply ui/src/app/components/help-sidebar.spec.tsx:532]: ok

**Review by jmthibault79 — APPROVED** [review]

**Pull request merged**